components/script/dom/gpudevice.rs

```diff
@@ -153,6 +155,86 @@ impl GPUDevice {
             )
         }
     }
+
+    fn validate_texture_view_binding(
```

Comment:
All the validation logic is common to the applications using wgpu. It needs to be implemented in `wgpu-core` instead of here. The idea is that you'd call wgpu-core API and get `Result`, from which you'd figure out how to throw an error according to the WebGPU error model.
We've started converting the methods to `Result`. As such, `create_bind_group_layout` returns one, as so do pipeline creation methods.

Therefore, I think the logic here needs to be removed.